src/checkoutservice/main.go

```diff
@@ -82,6 +89,43 @@ func InitTracerProvider() *sdktrace.TracerProvider {
 	return tp
 }
 
+func InitMeterProvider() (func(ctx context.Context) error, error) {
+	ctx := context.Background()
+
+	client := otlpmetricgrpc.NewClient(
+		otlpmetricgrpc.WithInsecure())
+
+	metricExporter, err := otlpmetric.New(ctx, client)
+	if err != nil {
+		log.Errorf("failed to create the metric collector exporter, %v", err)
+		return nil, err
+	}
+
+	controller := controller.New(
```

Comment:
Is this a good point in time to introduce a shared `resource` for the controller and the tracer provider?

Reply:
I think this may be better to be solved in another pr, since this also changed the behavior of trace?
@cedricziel @cartersocha

Reply:
resolving this for now. We can discuss more over slack